okpt/io/config/parsers/opensearch.py

```diff
@@ -35,9 +45,15 @@ class OpenSearchParser(base.BaseParser):
     def __init__(self):
         super().__init__('opensearch')
 
-    def parse(self, file_obj: TextIOWrapper) -> Dict[str, Any]:
+    def parse(self, file_obj: TextIOWrapper) -> OpenSearchConfig:
         """See base class."""
         config_obj = super().parse(file_obj)
         index_spec_path = config_obj['index_spec']
         index_spec_obj = reader.parse_json_from_path(index_spec_path)
-        return {**config_obj, 'index_spec': index_spec_obj}
+        opensearch_config = OpenSearchConfig(
+            index_spec=index_spec_obj,
+            max_num_segments=config_obj['max_num_segments'],
```

Comment:
Remind me: What happens if these keys are not in config_obj?

Reply:
`config_obj` is formed after validating the raw config through Cerberus, so if the config doesn't match the schema (e.g. missing keys), I throw a `ConfigurationError` during the validation process.

Reply:
Makes sense.